harness/determined/estimator/_callback.py

```diff
@@ -0,0 +1,29 @@
+import pathlib
+
+
+class EstimatorCallback:
+    """
+    Abstract base class used to define a callback that should execute during
+    the lifetime of a EstimatorCallback.
```

Comment:
blocking: add a warning that this callback should be stateless.

(in contrast, `PyTorchCallback` can specify how to save and load callback state and can be stateful)

Reply:
Users can save load/save callback state via `on_checkpoint_load` and `on_checkpoint_end` calls.